### DISCUSSION TITLE
Random Background

### DISCUSSION THREAD

**Discussion by VeriumETH** [nftchef/art-engine#94] [Help + Q&A] [answered]
Hello 👋 

I was wondering if the genColor and drawBackground constructors  can be setup to never use the same color code more than once?

**Comment by nftchef** [comment, marked as answer]
hi, it's _possible_ to do something like this, sure!
A possible approach would be to maintain a `new Set()` of colors, then each time the color is randomly selected, check first that it is not already in the Set [otherwise pick again], then add it to the set after using it. 

Although, the difference of even `1` for any RGB value would be considered unique, while _visually_ looking identical to another color. More care and finesse would need to be added to crates _steps_ of noticeably different colors
